### PR TITLE
Doc: Tree Infinite Scrolling

### PR DESCRIPTION
**Pull request by petdud**
## New Behavior

Adding an infinite scrolling example for the `Tree` component as requested by partners.

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Avatar-mount_base.html">587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Avatar-mount.html">578</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Button-mount_base.html">298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Button-mount.html">298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Field-mount_base.html">1021</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Field-mount.html">1025</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProvider-mount_base.html">653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProvider-mount.html">650</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-mount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">68</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">59</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/InfoButton-mount_base.html">8</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/MakeStyles-mount_base.html">848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/MakeStyles-mount.html">834</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Persona-mount_base.html">1603</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/Persona-mount.html">1557</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/SpinButton-mount_base.html">1241</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28197/perf-test-react-components/SpinButton-mount.html">1283</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `64.935 kB`<br />`17.918 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.151 kB`<br />`58.715 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.168 kB`<br />`11.964 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/36aacc5ff33fdb056c46a439da81a5c9b274e8ee'>36aacc5ff33fdb056c46a439da81a5c9b274e8ee</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          ]
+        : []),
+    ],
+    [isLoading, peopleItems],
+  );
+
+  const styles = useStyles();
+
+  const flatTree = useFlatTree_unstable(items, { defaultOpenItems: ['pinned', 'people'] });
+  const listRef = React.useRef<HTMLDivElement>(null);
```
Comment (bsunderhus) [inline packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx:62]: nit: This `listRef` is not being used.

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    const { scrollTop, scrollHeight, clientHeight } = event.currentTarget;
+    const hasReachedEnd = scrollHeight - scrollTop === clientHeight;
+
+    if (!isLoading && hasReachedEnd && page < MAX_PAGES) {
+      fetchMoreItems();
+    }
+  };
+
+  return (
+    <div ref={listRef} onScroll={handleScroll} className={styles.container}>
```
Comment (bsunderhus) [inline packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx:92]: Do we need this wrapper?! would it be the same if we just used `onScroll` and `className` directly on `Tree`?

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { Tree, TreeItem, TreeItemLayout, useFlatTree_unstable, FlatTreeItemProps } from '@fluentui/react-tree';
+import { makeStyles, shorthands } from '@fluentui/react-components';
+import story from './TreeInfiniteScrolling.md';
+import { Spinner } from '../../../react-spinner/src/Spinner';
```
Comment (bsunderhus) [inline packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx:5]: Let's import Spinner from `@fluentui/react-components`

**Review by bsunderhus — APPROVED** [review]
Besides some minor nits this looks great!

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx (hunk 1 above)
Comment (petdud) [reply packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx]: 🙈 removed

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx (hunk 2 above)
Comment (petdud) [reply packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx]: good point, it works fine with just the Tree, so wrapper removed 👍

File: packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx (hunk 3 above)
Comment (petdud) [reply packages/react-components/react-tree/stories/D_flatTree/TreeInfiniteScrolling.stories.tsx]: fixed 🙌

**Pull request merged**